balsamiq.md

```diff
@@ -0,0 +1,131 @@
+
+After the dust settled, we started imagining what we wanted to do for the next version and realized that our system had limitations that we would have to overcome. This is a story of three challenges and what we did to solve them.
+
+***Note:*** *These are specific to [Hugo](http://gohugo.io/), the static site generator we use, but should be adaptable to others.*
+
+## Challenge #1: Documenting multiple product versions
+
+Our product, [Balsamiq Mockups](https://balsamiq.com/), comes in multiple "flavors," as we often call them. There are two versions of the core product (the wireframe editor) that are sold as: a stand-alone desktop version, a hosted web version, and plugins for Google Drive and Atlassian Confluence and JIRA. 
+
+![](https://media.balsamiq.com/images/docslikecode/select-product.png)
+
+There are seven different products that we sell, catered to different types of users. This is great for our customers, but a challenge for our docs.
+
+The majority of our documentation is how to use one of the two core wireframe editors. Very little has to do with platform-specific features. So, seven products that we sell, but only two main sets of documentation. 
+
+For historical reasons, most of the common documentation is in the Desktop version docs. If you used our product on different platform, you had to know to go to the Desktop docs. 
+
+This is what the content folder in our [GitHub repo](https://github.com/balsamiq/docs.balsamiq.com) looked like, which meant that a lot of customers had this experience:
+
+![](https://media.balsamiq.com/images/docslikecode/structure2.png)
+
+For the next update to our docs, we decided that we needed to finally address this issue. We wanted a complete set of docs for each product version and platform.
+
+![](https://media.balsamiq.com/images/docslikecode/structure3.png)
+
+But, of course, we didn't want to maintain seven copies of each article. We wanted to be able to write the docs once and have them show up somehow across multiple versions.
+
+A simple solution, I suppose, would have been to switch from markdown to a file format that supports including content from one file in another (like [reStructuredText](http://docutils.sourceforge.net/rst.html) or [AsciiDoc](http://asciidoc.org/)). **But, I'm stubborn** (and Hugo doesn't support them well). I like how simple markdown is and I wanted to see if we could achieve what we wanted without abandoning it.
+
+Eventually, we came up with a way to do just what we wanted, creating articles for each core version of the product ("v2" and "v3") that would get inserted into files in multiple documentation folders.
+
```

Comment:
Ohhh I get it now, it's not really conditionals nor includes, but at-build-time file creation? Okay, one more sentence to connect the dots, then. "With the correct files in the correct locations, we could provide contextual information for the user in the location they rightfully expect."